examples/minwebgl/mapgen_tiles_rendering/src/main.rs

```diff
@@ -0,0 +1,305 @@
+//! Render tile map on quad.
+
+use gl::GL;
+use minwebgl as gl;
+use ndarray_cg::mat::DescriptorOrderColumnMajor;
+use web_sys::wasm_bindgen::prelude::*;
+
+const LAYERS: i32 = 6;
+const DATA: [u8; 256] = [
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 
+    0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 2, 1, 0,
+    0, 0, 0, 0, 0, 0, 0, 1, 2, 2, 1, 0, 1, 2, 2, 1, 
+    0, 0, 0, 0, 0, 0, 1, 2, 2, 2, 2, 1, 0, 1, 1, 0,
+    0, 0, 1, 1, 1, 1, 2, 2, 3, 3, 2, 2, 1, 0, 0, 0, 
+    0, 1, 2, 2, 1, 1, 2, 3, 4, 4, 3, 3, 2, 1, 0, 0,
+    1, 2, 3, 3, 2, 2, 2, 3, 4, 4, 4, 3, 2, 1, 0, 0, 
+    1, 2, 3, 3, 3, 3, 3, 4, 4, 4, 4, 3, 2, 1, 0, 0,
+    1, 2, 3, 4, 4, 4, 4, 4, 5, 5, 4, 3, 3, 2, 1, 0, 
+    1, 2, 3, 4, 4, 4, 4, 5, 5, 5, 4, 4, 3, 2, 1, 0,
+    1, 2, 3, 3, 4, 4, 1, 1, 5, 5, 4, 4, 3, 2, 1, 0, 
+    0, 1, 2, 3, 3, 1, 1, 4, 4, 4, 4, 3, 2, 1, 1, 0,
+    0, 0, 1, 2, 1, 1, 3, 3, 3, 3, 3, 3, 2, 1, 0, 0, 
+    0, 0, 0, 1, 1, 2, 2, 2, 2, 2, 2, 2, 1, 0, 0, 0,
+    0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+];
+
+fn set_resize_callback() {
+    let on_resize_callback = move |canvas: &web_sys::HtmlCanvasElement| {
+        let gl = gl::context::retrieve_or_make().expect("Should have a context");
+        let window = web_sys::window().expect("Should have a window");
+        let inner_width = window.inner_width().unwrap().as_f64().unwrap() as u32;
+        let inner_height = window.inner_height().unwrap().as_f64().unwrap() as u32;
+        canvas.set_width(inner_width);
+        canvas.set_height(inner_height);
+        gl.viewport(0, 0, inner_width as i32, inner_height as i32);
+        update();
+    };
+
+    let gl = gl::context::retrieve_or_make().expect("Should have a context");
+    let on_resize_callback: Box<dyn Fn(&web_sys::HtmlCanvasElement)> = Box::new(on_resize_callback);
+    let canvas = gl
+        .canvas()
+        .expect("Canvas should exist")
+        .dyn_into::<web_sys::HtmlCanvasElement>()
+        .unwrap();
+    on_resize_callback(&canvas);
+    let on_resize_callback: Closure<dyn Fn()> = Closure::new(move || on_resize_callback(&canvas));
+    let window = web_sys::window().expect("Should have a window");
+    window.set_onresize(Some(on_resize_callback.as_ref().unchecked_ref()));
+    on_resize_callback.forget();
+}
+
+fn set_load_callback() {
+    let load = move |_img: &web_sys::HtmlImageElement| {
+        update();
+    };
+
+    let _ = load_image("tileset.png", Box::new(load));
+    let _ = load_image("map.png", Box::new(load));
+}
+
+fn load_image(
```

Comment:
We have function for that in mingl, you can check it [here](https://github.com/Wandalen/cgtools/blob/55107fbe36c68ef310fb23ca21d45dc3668dc824/module/min/mingl/src/web/dom.rs#L32).